src/base/bittorrent/infohash.cpp

```diff
@@ -89,5 +89,9 @@ bool BitTorrent::operator!=(const InfoHash &left, const InfoHash &right)
 
 uint BitTorrent::qHash(const InfoHash &key, const uint seed)
 {
+#if (LIBTORRENT_VERSION_NUM < 10200)
     return ::qHash(static_cast<QString>(key), seed);
+#else
+    return ::qHash((std::hash<lt::sha1_hash> {})(key), seed);
```

Comment:
Is it really more efficient/fast? You do use two hashing here!

Reply:
>Is it really more efficient/fast? 

IMO yes, please see the commit message. I can dig out the implementation details if you want.

>You do use two hashing here!

I think both ways should be equally good (in terms of functionality), but in terms of performance I think one is better than the other and it makes sense to pick the faster one (the number of hashing functions invoked is not the real issue here IMO).

Reply:
>You do use two hashing here!

At first I considered using `std::hash(key)` and mix with `seed` ourselves but then I thought why do it ourselves when Qt already provides a overload of `qHash()` that ensures `seed` is handled properly and integers larger than `uint` is narrowed properly.